src/v/transform/transform_manager.cc

```diff
@@ -0,0 +1,386 @@
+    _queue.submit(
+      [this, id, ntp = std::move(ntp), meta = std::move(meta)]() mutable {
+          return handle_transform_error(id, std::move(ntp), std::move(meta));
+      });
+}
+
+ss::future<>
+manager::handle_leadership_change(model::ntp ntp, ntp_leader leader_status) {
+    vlog(
+      tlog.debug,
+      "handling leadership status change to leader={} for: {}",
+      leader_status,
+      ntp);
+
+    if (leader_status == ntp_leader::no) {
+        // We're not the leader anymore, time to shutdown all the processors
+        co_await _processors->erase_by_ntp(ntp);
+        co_return;
+    }
+    // We're the leader - start all the processor that aren't already running
+    auto transforms = _registry->lookup_by_input_topic(
+      model::topic_namespace_view(ntp));
+    co_await ss::parallel_for_each(
+      std::move(transforms),
+      [this, &ntp](auto entry) { return start_processor(ntp, entry.first); });
+}
+
+ss::future<> manager::handle_plugin_change(model::transform_id id) {
+    vlog(tlog.debug, "handling update to plugin: {}", id);
+    // If we have an existing processor we need to restart it with the updates
+    // applied.
+    co_await _processors->erase_by_id(id);
+
+    auto transform = _registry->lookup_by_id(id);
+    // If there is no transform we're good to go, everything is shutdown if
+    // needed.
+    if (!transform) {
+        co_return;
+    }
+    // Otherwise, start a processor for every partition we're a leader of.
+    auto partitions = _registry->get_leader_partitions(transform->input_topic);
+    co_await ss::parallel_for_each(
+      partitions.begin(),
+      partitions.end(),
+      [this, id, transform = *std::move(transform)](
+        model::partition_id partition) {
+          auto ntp = model::ntp(
+            transform.input_topic.ns, transform.input_topic.tp, partition);
+          // It's safe to directly create processors, because we deleted them
+          // for a full restart from this deploy
+          return create_processor(std::move(ntp), id, transform);
+      });
+}
+
+ss::future<> manager::handle_transform_error(
+  model::transform_id id, model::ntp ntp, model::transform_metadata meta) {
+    auto* entry = _processors->get_or_null(id, ntp);
+    if (!entry) {
+        co_return;
+    }
+    co_await entry->processor()->stop();
+    auto delay = entry->backoff()->next_backoff_duration();
+    vlog(
+      tlog.info,
+      "transform {} errored on partition {}, delaying for {} then restarting",
+      meta.name,
+      ntp.tp.partition,
+      delay);
+    _queue.submit_delayed(delay, [this, id, ntp]() mutable {
+        return start_processor(std::move(ntp), id);
+    });
+}
+
+ss::future<> manager::start_processor(model::ntp ntp, model::transform_id id) {
+    auto* entry = _processors->get_or_null(id, ntp);
+    // It's possible something else came along and kicked this processor and
+    // that worked.
+    if (entry && entry->processor()->is_running()) {
```

Comment:
i see how it's used now the processor lifecycle is controlled by this manager. that's cool.

if a processor is created (but not started) or stopped, _task = ss::now() occurs, and i think that would evaluate to not running, so then we'd fall through and start it. that seems to check out.